modernizer-maven-plugin/src/test/java/org/gaul/modernizer_maven_plugin/ModernizerSuppressionsTest.java

```diff
@@ -0,0 +1,287 @@
+    .IgnoreConstructorTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreGenericClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreGenericClassConstructorTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodInGenericClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningArrayClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningArrayPrimitiveTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningDeclaredTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningGenericTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodReturningPrimitiveTypeClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithArrayTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithDeclaredTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithEmptyParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithGenericTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithPrimitiveAndGenericTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithPrimitiveTypeParametersTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreMethodWithVoidParameterTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreOverloadedMethodInGenericClassTest;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreOverloadedMethodTestClass;
+import org.gaul.modernizer_maven_plugin.ModernizerTestHelper
+    .IgnoreStaticOuterClassConstructor.IgnoreStaticInnerClassConstructor;
+import org.junit.Before;
+import org.junit.Test;
+import org.objectweb.asm.ClassReader;
+
+public final class ModernizerSuppressionsTest {
+    private Map<String, Violation> violations;
+    private static final Collection<String> NO_EXCLUSIONS =
+        Collections.<String>emptySet();
+    private static final Collection<Pattern> NO_EXCLUSION_PATTERNS =
+        Collections.<Pattern>emptySet();
+    private static final Collection<String> NO_IGNORED_PACKAGES =
+        Collections.<String>emptySet();
+    private static final Collection<String> NO_IGNORED_METHODS =
+        Collections.<String>emptySet();
+    private List<Pattern> ignoreClasses = new ArrayList<Pattern>();
+    private List<String> ignoreMethods = new ArrayList<String>();
+
+    @Before
+    public void setUp() throws Exception {
+        violations = ModernizerTestUtils.readViolations();
+        String currentDirectory = System.getProperty("user.dir");
```

Comment:
Is this code that can / should be shared?

Reply:
Reading the ignore files happens in `ModernizerMojo` - which is the main source code. So, we would need to move it to `Utils` and then call the method here. But, not sure if its best to move the original plugin code.

Reply:
We can leave it for now